osf/models/queued_mail.py

```diff
@@ -88,6 +91,8 @@ def queue_mail(to_addr, mail, send_at, user, **context):
                     not parameters.
     :return: the QueuedMail object created
     """
+    if waffle.switch_is_active(DISABLE_ENGAGEMENT_EMAILS) and mail in [NO_LOGIN, NO_ADDON, NEW_PUBLIC_PROJECT, WELCOME_OSF4M]:
```

Comment:
We don't want to prevent the [`PREREG_REMINDER`](https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/queued_mail.py#L132) email from being sent. We could simplify this to: 
```
if waffle.switch_is_active and mail is not PREREG_REMINDER:
	return False 
```
but that feels a bit less explicit.

Reply:
Oh, I see. It's a bit smelly to have these hardcoded here. What if you add a key to each of the disable-able mails? Something like

```diff
diff --git a/osf/models/queued_mail.py b/osf/models/queued_mail.py
index 7b34f702af..56896fd93c 100644
--- a/osf/models/queued_mail.py
+++ b/osf/models/queued_mail.py
@@ -91,7 +91,7 @@ def queue_mail(to_addr, mail, send_at, user, **context):
                     not parameters.
     :return: the QueuedMail object created
     """
-    if waffle.switch_is_active(DISABLE_ENGAGEMENT_EMAILS) and mail in [NO_LOGIN, NO_ADDON, NEW_PUBLIC_PROJECT, WELCOME_OSF4M]:
+    if waffle.switch_is_active(DISABLE_ENGAGEMENT_EMAILS) and mail.get('engagement', False):
         return False
     new_mail = QueuedMail(
         user=user,
@@ -117,6 +117,7 @@ NO_ADDON = {
     'template': 'no_addon',
     'subject': 'Link an add-on to your OSF project',
     'presend': presends.no_addon,
+    'engagement': True,
     'categories': ['engagement', 'engagement-no-addon']
 }
 
```